### PR TITLE
net: zperf: Make shell dependency optional

### PR DESCRIPTION
**Pull request by krish2718**
Now that we a proper API, shell is just optional, so, make the dependency optional by refactoring the code.

Signed-off-by: Krishna T <krishna.t@nordicsemi.no>

### CODE REVIEW

File: samples/net/zperf/sample.yaml [hunk 1]
```diff
@@ -10,6 +10,10 @@ sample:
 tests:
   sample.net.zperf:
     platform_allow: qemu_x86
+  sample.net.zperf_no_shell:
+    extra_configs:
+      - CONFIG_NET_SHELL=n
```
Comment (jukkar) [inline samples/net/zperf/sample.yaml:15]: There are mixed dependencies about the shell.
Here we set `CONFIG_NET_SHELL` but in various settings we depend on `CONFIG_SHELL` which imho looks wrong. As this is network library we should depend on `CONFIG_NET_SHELL` only (it will depend on `CONFIG_SHELL`)

File: subsys/net/lib/zperf/CMakeLists.txt [hunk 2]
```diff
@@ -5,13 +5,16 @@ zephyr_library_named(zperf)
...
   zperf_common.c
   zperf_session.c
-  zperf_shell.c
   zperf_udp_receiver.c
   zperf_udp_uploader.c
   zperf_tcp_receiver.c
   zperf_tcp_uploader.c
 )
 
+zephyr_library_sources_ifdef(CONFIG_SHELL
```
Comment (jukkar) [inline subsys/net/lib/zperf/CMakeLists.txt:14]: prefer `CONFIG_NET_SHELL`

**Review by jukkar — DISMISSED** [review]
LGTM

File: subsys/net/lib/zperf/zperf_shell.c [hunk 3]
```diff
@@ -1,11 +1,11 @@
-/*
+	/*
```
Comment (jukkar) [inline subsys/net/lib/zperf/zperf_shell.c:1]: Weird and unnecessary change here

**Review by jukkar — COMMENTED** [review]
Minor nit.
BTW, if you are sending a new version after people have reviewed the code already, it would be nice to get some info in the comment what has changed. This helps the reviewer to concentrate to the changed parts.

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
LGTM, thanks for the cleanup!
